flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterGateway.java

```diff
@@ -316,4 +316,11 @@ CompletableFuture<?> stopTrackingAndReleasePartitions(
      */
     CompletableFuture<Acknowledge> updateJobResourceRequirements(
             JobResourceRequirements jobResourceRequirements);
+
+    /**
+     * Notifies that the execution has reached the end of data and is to be finished.
+     *
+     * @param executionAttempt The execution attempt id.
+     */
+    void notifyTaskFinishing(final ExecutionAttemptID executionAttempt);
```

Comment:
Would it be a bit more simpler to name this method `notifyEndOfData`? And update the doc to remove `is to be finished`? The reason is that we do not need to additionally define what "finishing task" means.